src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.dao.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.Role.RoleName;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
+    private static final String USER_EMAIL = "bob@com.ua";
+    private static final String USER_PASSWORD = "password";
+    private static final String EMAIL_NON_EXIST = "nonexist@com.ua";
+    private UserDao userDao;
+    private User user;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {User.class, Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        RoleDao roleDao = new RoleDaoImpl(getSessionFactory());
+        Role roleUser = new Role();
+        roleUser.setRoleName(RoleName.USER);
+        roleDao.save(roleUser);
+        user = new User();
+        user.setEmail(USER_EMAIL);
+        user.setPassword(USER_PASSWORD);
+        user.setRoles(Set.of(roleUser));
+    }
+
+    @Test
+    void save_Ok() {
```

Comment:
I think is a little overhead.
You can just compare users and if user is saved he have non null id.